tests/testthat/test-shinytest2-teal.R

```diff
@@ -0,0 +1,317 @@
+      rvest::read_html() %>%
+      rvest::html_text(),
+    app_title
+  )
+  expect_equal(
+    app$get_html("head > link[rel='icon']") %>%
+      rvest::read_html() %>%
+      rvest::html_elements("link") %>%
+      rvest::html_attr("href"),
+    app_favicon
+  )
+  expect_true(
+    grepl(
+      app_header,
+      app$get_html("header") %>%
+        rvest::read_html() %>%
+        rvest::html_text()
+    )
+  )
+  expect_true(
+    grepl(
+      app_footer,
+      app$get_html("footer") %>%
+        rvest::read_html() %>%
+        rvest::html_text()
+    )
+  )
+  app$stop()
+})
+
+test_that("expected teal filters are initialized when module specific filters are created", {
+  # App with module specific filters
+  app <- TealAppDriver$new(
+    data = simple_teal_data(),
+    modules = modules(
+      example_module(label = "Module_1"),
+      example_module(label = "Module_2")
+    ),
+    filter = teal_slices(
+      teal_slice(id = "iris_species", dataname = "iris", varname = "Species", multiple = TRUE),
+      teal_slice(id = "mtcars_cyl", dataname = "mtcars", varname = "cyl", selected = c(4, 6)),
+      teal_slice(id = "mtcars_drat", dataname = "mtcars", varname = "drat", selected = c(3, 4)),
+      teal_slice(id = "mtcars_gear", dataname = "mtcars", varname = "gear"),
+      module_specific = TRUE,
+      mapping = list(
+        "Module_1" = c("iris_species", "mtcars_cyl"),
+        "Module_2" = c("iris_species", "mtcars_drat", "mtcars_gear")
+      )
+    )
+  )
+  app$wait_for_idle(timeout = default_idle_timeout)
+
+  expect_identical(app$get_active_data_filters("iris"), "Species")
+  expect_identical(app$get_active_data_filters("mtcars"), "cyl")
+  expect_identical(
+    app$get_filter_selection_value("iris", "Species"),
+    c("setosa", "versicolor", "virginica")
+  )
+  expect_identical(
+    app$get_filter_selection_value("mtcars", "cyl"),
+    c("4", "6")
+  )
+  expect_null(app$get_filter_selection_value("mtcars", "drat", is_numeric = TRUE))
+  expect_null(app$get_filter_selection_value("mtcars", "gear"))
+
+  app$navigate_teal_tab("Module_2")
+  app$wait_for_idle(timeout = default_idle_timeout)
+
+  expect_identical(app$get_active_data_filters("iris"), "Species")
+  expect_identical(app$get_active_data_filters("mtcars"), c("drat", "gear"))
+  expect_identical(
+    app$get_filter_selection_value("iris", "Species"),
+    c("setosa", "versicolor", "virginica")
+  )
+  expect_identical(
+    app$get_filter_selection_value("mtcars", "drat", is_numeric = TRUE),
+    c(3, 4)
+  )
+  expect_identical(
+    app$get_filter_selection_value("mtcars", "gear"),
+    c("3", "4", "5")
+  )
+  expect_null(app$get_filter_selection_value("mtcars", "cyl"))
+
+  app$set_filter_selection_value("iris", "Species", "setosa")
+  app$navigate_teal_tab("Module_1")
+  app$wait_for_idle(timeout = default_idle_timeout)
+
+  expect_identical(
+    app$get_filter_selection_value("iris", "Species"),
+    "setosa"
+  )
+  app$stop()
+})
+
```

Comment:
There are following missing tests but descriptions might be slightly different. Generally items in module-specific filter panel are independent, but some filters can be shared. If filter is mapped into multiple modules then it is the same filter (same object, same pointer). This means that changing values in one is reflected in another. 
It is also possible to add a new filter with the same variable/values and map it to single module - then the other (theoretically duplicated) filter is not affected. 

testthat::test_that("setting values of shared module-specific items are reflected in all modules", {})
testthat::test_that("setting values of independent module-specific items are limit only to this module", {})

Reply:
Thanks, this is done in 66dfc495e5c24df7ed39202944cf80faab80affd